homeassistant/components/here_weather/weather.py

```diff
@@ -0,0 +1,222 @@
+"""Weather platform for the HERE Destination Weather service."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.weather import (
+    ATTR_FORECAST_CONDITION,
+    ATTR_FORECAST_PRECIPITATION,
+    ATTR_FORECAST_TEMP,
+    ATTR_FORECAST_TEMP_LOW,
+    ATTR_FORECAST_TIME,
+    ATTR_FORECAST_WIND_BEARING,
+    ATTR_FORECAST_WIND_SPEED,
+    WeatherEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import (
+    CONDITION_CLASSES,
+    DEFAULT_MODE,
+    DOMAIN,
+    MODE_ASTRONOMY,
+    MODE_DAILY_SIMPLE,
+    SENSOR_TYPES,
+)
+from .utils import convert_unit_of_measurement_if_needed, get_attribute_from_here_data
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry, async_add_entities
+):
+    """Add here_weather entities from a config_entry."""
+    here_weather_data_dict = hass.data[DOMAIN][config_entry.entry_id]
+    for sensor_type in SENSOR_TYPES:
+        if sensor_type != MODE_ASTRONOMY:
+            async_add_entities(
```

Comment:
Please only call add entities once with all entities.

Reply:
Implemented in https://github.com/home-assistant/core/pull/28910/commits/5669b5dbf2f5d06db7b5e48a30596f85019be74c